src/lsstseries/analysis/structurefunction2.py

```diff
@@ -81,29 +87,82 @@ def calc_sf2(time, flux, err=None, band=None, lc_id=None, sf_method="basic", arg
 
             # Create stacks of critical quantities, indexed by id
             id_masks = [lc_ids == lc for lc in unq_ids]
-            times_2d = [times[mask] for mask in id_masks]
-            fluxes_2d = [fluxes[mask] for mask in id_masks]
-            errors_2d = [errors[mask] for mask in id_masks]
-
-            sf_calculator = SF_METHODS[sf_method](times_2d, fluxes_2d, errors_2d, argument_container)
-
-            res = sf_calculator.calculate()
 
-            res_ids = [[str(unq_ids[i])] * len(arr) for i, arr in enumerate(res[0])]
-            res_bands = [[b] * len(arr) for arr in res[0]]
+            lightcurves = []
+            for mask in id_masks:
+                sf_lc = StructureFunctionLightCurve(
+                    times=times[mask], fluxes=fluxes[mask], errors=errors[mask]
+                )
+                lightcurves.append(sf_lc)
+
+            sf_calculator = SF_METHODS[sf_method](lightcurves, argument_container)
+
+            # aggregated_* has shape [calc_rep(0:arg_container.calc_repetitions)][lc_id(0:num_lightcurves)][bin(0:num_dt_bins)]
```

Comment:
Is this a relic of some previous code that was not deleted?

Reply:
This is meant to be a comment indicating the resulting shape of the following two variables. I found it challenging to keep track of what each of the dimensions represented, so I wanted to make a note of it here. However, if it's not intuitive, then I can rework this to make it more clear.